File: 35-search-insert-position/search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        low = 0
        high = len(nums)-1
        while(low<high):
            mid = (high+low)//2
            if nums[mid] == target:
                return mid
            elif nums[mid]<target:
                low = mid+1
            else:
                high = mid -1
        if high<0:
            high=0
        if nums[high]>=target:
            return high
        else:
            return high+1

File: 35-search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int high=nums.length-1,low=0;
        while(low<=high)
        {
            int mid = (low+high)/2;
            if(nums[mid] == target)
            {
                return mid;
            }
            else if(nums[mid]>target)
            {
                high = mid -1;
            }
            else
            {
                low = mid+1;
            }
        }
        return high+1;
    }
}